packages/eslint-plugin/src/rules/explicit-function-return-type.ts

```diff
@@ -65,6 +65,17 @@ export default util.createRule<Options, MessageIds>({
       );
     }
 
+    /**
+     * Checks if the parent of a function expression has a type annotation.
+     * @param parent The parent of a function expression node
+     */
+    function hasTypeAnnotation(parent: TSESTree.Node): boolean {
+      return (
+        parent.type === AST_NODE_TYPES.VariableDeclarator &&
+        'typeAnnotation' in parent.id
```

Comment:
Better to double check that there is actually a type annotation value rather than just checking for the property existence.

```suggestion
        'typeAnnotation' in parent.id &&
        !!parent.id.typeAnnotation
```